polaris-react/src/components/PolarisTestProvider/PolarisTestProvider.tsx

```diff
@@ -66,7 +66,10 @@ export function PolarisTestProvider({
   const stickyManager = useMemo(() => new StickyManager(), []);
 
   const featuresConfig = useMemo(
-    () => ({...features, polarisSummerEditions2023: false}),
+    () => ({
+      polarisSummerEditions2023: false,
+      ...features,
+    }),
```

Comment:
Swapping the order to make sure the mount function used on tests can toggle the value of `polarisSummerEditions2023`